tests/slurm-tests/asr_nim/run_test.py

```diff
@@ -0,0 +1,420 @@
+        raise ValueError(f"Configuration file {config_file} does not contain 'nim_configurations' dict")
+    
+    nim_configurations = config_module.nim_configurations
+    
+    if config_key not in nim_configurations:
+        raise ValueError(
+            f"Configuration key '{config_key}' not found. "
+            f"Available keys: {list(nim_configurations.keys())}"
+        )
+    
+    return nim_configurations[config_key]
+
+
+def get_container_path(cluster, nim_config):
+    """Return appropriate container path based on cluster and NIM config.
+    
+    Args:
+        cluster: Cluster name
+        nim_config: NIM configuration dict
+        
+    Returns:
+        str: Container image path
+    """
+    # Check if there's a local image for this cluster
+    local_images = nim_config.get("local_images", {})
+    if cluster in local_images:
+        return local_images[cluster]
+    
+    # Fall back to remote image URI
+    return nim_config["image_uri"]
+
+
+def setup_workspace_and_mounts(workspace, cluster):
+    """Create workspace directory and generate mount paths.
+    
+    Args:
+        workspace: Workspace directory path
+        cluster: Cluster name
+        
+    Returns:
+        tuple: (cluster_config, mount_paths)
+    """
+    from nemo_skills.pipeline.utils import get_cluster_config, create_remote_directory
+    
+    # Get cluster config and create workspace directory
+    cluster_config = get_cluster_config(cluster)
+    create_remote_directory(workspace, cluster_config)
+    
+    # Generate mount paths for workspace and HF_HOME
+    workspace_parent = str(Path(workspace).parent)
+    
+    # Extract HF_HOME from env_vars
+    hf_home = None
+    env_vars = cluster_config.get("env_vars", [])
+    for env_var in env_vars:
+        if isinstance(env_var, str) and "HF_HOME=" in env_var:
+            hf_home = env_var.split("=", 1)[1]
+            break
+    
+    if not hf_home:
+        hf_home = "/lustre/fsw/portfolios/convai/users/vmendelev/.cache/huggingface"
+    
+    mount_paths = f"{workspace_parent}:{workspace_parent},{hf_home}:{hf_home}"
+    
+    return cluster_config, mount_paths
+
+
+def prepare_test_data(workspace, cluster):
+    """Upload test data and audio files to cluster, fix paths in JSONL.
+    
+    Args:
+        workspace: Workspace directory path
+        cluster: Cluster name
+        
+    Returns:
+        str: Path to the prepared input file on the cluster
+    """
+    from nemo_skills.pipeline.utils.cluster import cluster_upload, get_tunnel
+    from nemo_skills.pipeline.utils import get_cluster_config, create_remote_directory
+    
+    # Get cluster config
+    cluster_config = get_cluster_config(cluster)
+    
+    # Ensure workspace exists
+    create_remote_directory(workspace, cluster_config)
+    
+    # Get tunnel and establish connection
+    tunnel = get_tunnel(cluster_config)
+    tunnel.run("echo 'Establishing connection...'", hide=True)
+    
+    # Upload test data JSONL
```

Comment:
I don't think we need an explicit upload here. Anything inside this repo will be packaged and uploaded automatically when job is submitted so you just need to reference those as `/nemo_run/code/tests/slurm-tests/asr_nim/wavs/...`